### PR TITLE
[warm reboot]Junyi single warmreboot

### PR DESCRIPTION
**Pull request by ms-junyi**
why you did it/What is the motivation for this PR?
Support warmreboot verification

How did you do it?
1. setup, teardown remain the same as before.Add a decorator to runtest to define the behavior of warmreboot
2. Establish a communication mechanism with sonic-mgmt

How did you verify/test it?
1. verify the warm-reboot cases(vlan and fdb) on dut s6000
3. Tested bcrm cases with pipeline to avoid affecting other cases on dut s6000

### CODE REVIEW

File: test/sai_test/sai_fdb_test.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def setUp(self):
         T0TestBase.setUp(self, is_reset_default_vlan=False)
         
 
+    @warm_test(is_runTest=True)
```
Comment (richardyu-ms) [inline test/sai_test/sai_fdb_test.py:42]: is_runTest => is_test_rebooting?

File: test/sai_test/sai_utils.py [hunk 2]
```diff
@@ -178,3 +178,50 @@ def generate_ip_address_list(role, group, indexes):
     for index in indexes:
         ip_list.append(role.format(group, index))
     return ip_list
+
+def warm_test(is_runTest:bool=False, time_out=60, interval=1):
```
Comment (richardyu-ms) [inline test/sai_test/sai_utils.py:182]: please add description for each parameter

File: test/sai_test/sai_utils.py [hunk 3]
```diff
@@ -178,3 +178,50 @@ def generate_ip_address_list(role, group, indexes):
...
+			# msg to promote change the text file content to rebooted
+                warm_file = open('/tmp/warm_reboot','w+')
+				# write content to reboot-requested
+                times = 0
+                print("write rebooting to file")
+                warm_file.write(WARM_TEST_REBOOTING)
+                warm_file.close()
+                try:
+                    while 1:
+                        warm_file = open('/tmp/warm_reboot','r')
```
Comment (richardyu-ms) [inline test/sai_test/sai_utils.py:206]: please print message when entring into each condition, with the reason and what's the effect.

File: test/sai_test/constant.py [hunk 4]
```diff
@@ -53,3 +53,7 @@
 SERVER_IPV6_PATTERN = 'fc02::{}:{}'
 T0_IPV6_PREFIX = 'fc00:0::{}:{}'
 T1_IPV6_PATTERN = 'fc00:1::{}:{}'
+
```
Comment (richardyu-ms) [inline test/sai_test/constant.py:56]: add some comment for the purpose of those variable

**Comment by rlhui** [comment]
Please update PR title to remove "Junyi"? Thanks.  Can we please also update the PR title to reflect more the change, as "single warmboot" test was there before?

**Comment by ms-junyi** [comment]
close pr due to no sign off in two commits

**Comment by richardyu-ms** [comment]
Raised new PR https://github.com/opencomputeproject/SAI/pull/1668

**Comment by richardyu-ms** [comment]
> Please update PR title to remove "Junyi"? Thanks. Can we please also update the PR title to reflect more the change, as "single warmboot" test was there before?

changed in another PR https://github.com/opencomputeproject/SAI/pull/1668

**Pull request closed without merging**